### ISSUE TITLE
datetime dumping to json flips sign of timezone

### ISSUE THREAD

**Issue by FynnBe** [pydantic/pydantic#8964]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

`TypeAdapter(datetime).dump_python(isoparse("2024-03-06T14:21:34-00:08"), mode="json")` flips the time zone sign to '+'

### Example Code

```Python
from datetime import datetime

from dateutil.parser import isoparse
from pydantic import TypeAdapter

iso_str = "2024-03-06T14:21:34-00:08"
expected = isoparse(iso_str)

datetime_adapter = TypeAdapter(datetime)

actual_datetime = datetime_adapter.validate_python(iso_str)
assert actual_datetime == expected

python_dump = datetime_adapter.dump_python(actual_datetime, mode="python")
assert python_dump == expected, (python_dump, expected)

json_dump = datetime_adapter.dump_python(actual_datetime, mode="json")
assert json_dump == iso_str, (
    json_dump,
    iso_str,
)  # fails with ('2024-03-06T14:21:34+00:08', '2024-03-06T14:21:34-00:08')
#                                    ^                            ^
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.6.3
        pydantic-core version: 2.16.3
          pydantic-core build: profile=release pgo=false
                 install path: C:\Users\fbeut\mambaforge\envs\bio38\Lib\site-packages\pydantic
               python version: 3.8.17 | packaged by conda-forge | (default, Jun 16 2023, 07:01:59) [MSC v.1929 64 bit (AMD64)]
                     platform: Windows-10-10.0.19041-SP0
             related packages: email-validator-2.0.0.post2 pydantic-settings-2.1.0 pyright-1.1.352 typing_extensions-4.7.1
                       commit: unknown
```

**Comment by sydney-runkle** [comment]
@FynnBe,

Thanks for reporting! Indeed, this is a bug. Probably needs to be fixed in `pydantic-core`. I'll take a look today!

**Comment by sydney-runkle** [comment]
Fixed this in `speedate`, will be released with `pydantic v2.7.0`!
